### DISCUSSION TITLE
Does it make sense to package the edge releases for openSUSE (instead of following stable)?

### DISCUSSION THREAD

**Discussion by johanneskastl** [linkerd/linkerd2#12669] [General]
Hi all,

I have been packaging linkerd-cli for openSUSE and was following the stable release. Now it seems there will no longer be any stable releases from the project directly.

Does it make sense to follow the edge releases with the package and drop the stable package? Or are there plans to make stable CLI releases each quarter or similar?

Kind Regards,
Johannes

**Comment by kastl-ars** [comment]
No opinions, anyone?

**Comment by wmorgan** [comment]
Hi @johanneskastl @kastl-ars, sorry about the delay. Not sure how I missed this question back in May.

First, thank you for your work in packaging Linkerd for OpenSUSE. On behalf of the Linkerd maintainers, that is much appreciated, and if you ever have reason to attend a Kubecon please come say hi and let us buy you a coffee/tea/beer/water/firm handshake.

Here's my suggestion. Each edge release is intended for production use and is thoroughly tested (including by Buoyant, which runs it in their production environment). Bugs happen, of course, and if we find an issue with a specific edge release we update [the release notes](https://github.com/linkerd/linkerd2/releases) to mark that edge release as "not recommended". But keeping up with every edge release seems like overkill. So I would basically pick a cadence that feels comfortable to you (and to OpenSUSE users), e.g. monthly or quarterly, and at that cadence, find the the latest edge release that is marked as recommended, and package it.

Additionally, when we announce a major Linkerd version (e.g. the upcoming 2.16) we also announce the edge release that corresponds to that version (e.g. [you can see that edge-24.2.4 corresponds to Linkerd 2.15](https://linkerd.io/releases/#linkerd-215)). You may also want to package those, or you could simply wait until the cadence comes around.

I appreciate that the new setup makes things more complicated for you than before. Sorry about that. Please know that it has dramatically improved our ability to maintain and improve Linkerd.

Let me know what you think.

**Reply by kastl-ars** [reply]
Thanks for the reply, William!

> First, thank you for your work in packaging Linkerd for OpenSUSE. On behalf of the Linkerd maintainers, that is much appreciated, and if you ever have reason to attend a Kubecon please come say hi and let us buy you a coffee/tea/beer/water/firm handshake.

:-)

> Additionally, when we announce a major Linkerd version (e.g. the upcoming 2.16) we also announce the edge release that corresponds to that version (e.g. [you can see that edge-24.2.4 corresponds to Linkerd 2.15](https://linkerd.io/releases/#linkerd-215)). You may also want to package those, or you could simply wait until the cadence comes around.

Sticking with the edge release relating to a major version sounds good. Not sure if there are any important updates after that, e.g. security fixes that should be included?

I'll start with packaging only the "major" edge release and see if this works out. If it is not that much of a hassle maybe I can package the current release in addition.

Have a nice day!

Kind Regards,
Johannes
